sdc/crypto/scripts/generate_secrets.py

```diff
@@ -0,0 +1,184 @@
+    :param keys_folder: The location on disk where the key exists
+    :param kid_override: This allows the caller to override the generated KID value
+    :return: None
+    '''
+    public_key_data = get_file_contents(keys_folder, public_key)
+
+    _generate_kid_from_key_and_add_to_dict(keys, "public", platform, service, purpose, key_use, version,
+                                           public_key_data, kid_override=kid_override)
+
+
+def add_private_key_to_dict(keys, platform, service, purpose, key_use, version, private_key, keys_folder,
+                            kid_override=None):
+    '''
+    Loads a private key from the file system and adds it to a dict of keys
+    :param keys: A dict of keys
+    :param platform the platform the key is for
+    :param service the service the key is for
+    :param key_use what the key is used for
+    :param version the version of the key
+    :param purpose: The purpose of the private key
+    :param private_key: The name of the private key to add
+    :param keys_folder: The location on disk where the key exists
+    :param kid_override: This allows the caller to override the generated KID value
+    :return: None
+    '''
+    private_key_data = get_file_contents(keys_folder, private_key)
+
+    private_key = load_pem_private_key(private_key_data.encode(), None, backend=backend)
+
+    pub_key = private_key.public_key()
+
+    pub_bytes = pub_key.public_bytes(Encoding.PEM, PublicFormat.SubjectPublicKeyInfo)
+
+    _generate_kid_from_key_and_add_to_dict(keys, "private", platform, service, purpose, key_use, version,
+                                           pub_bytes.decode(), private_key_data, kid_override=kid_override)
+
+
+def generate_secrets_file(keys, ):
+    with open('secrets.yml', 'w') as f:
+        yaml.dump({"keys": keys}, f, default_flow_style=False)
+
+
+def generate_keys(keys_folder):
+    key_files = [f for f in os.listdir(keys_folder) if os.path.isfile(os.path.join(keys_folder, f))]
+
+    keys = {}
+
+    for key_file in key_files:
+        try:
+            # remove the .pem extension
+            key_file_no_pem = key_file[:key_file.index(".")]
+            platform, service, purpose, key_use, key_type, version = key_file_no_pem.split("-")
+            if key_type == "public":
+                add_public_key_to_dict(keys, platform, service, purpose, key_use, version, key_file, keys_folder)
+            elif key_type == "private":
+                add_private_key_to_dict(keys, platform, service, purpose, key_use, version, key_file, keys_folder)
+            else:
+                print("Unknown key type {} in {}".format(key_type, key_file))
+
+        except ValueError:
+            print("File {} is not in correct format".format(key_file))
+
+    generate_secrets_file(keys)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Generate secrets key file.')
+    parser.add_argument('folder', type=str,
```

Comment:
Yes agreed, actually I'll add more output so it's easy to see which files have been processed and what file has been generated.